spacy/compat.py

```diff
@@ -27,6 +27,13 @@
 except ImportError:
     from typing_extensions import Literal  # noqa: F401
 
+
+try:  # Python 3.8+
+    import importlib.metadata as importlib_metadata
+except ImportError:
+    from catalogue import _importlib_metadata as importlib_metadata  # noqa: F401
+
+
```

Comment:
I don't think the kind of compat is a good idea because `packages_distributions` is backport from a pretty recent version of `importlib_metadata` and I don't think it's necessarily in `importlib.metadata`, if at all. This definitely needs some cross-python tests.

Reply:
Oh, I was under the impression that `packages_distributions` was always part of `importlib.metadata` in Python 3.8 and above, but only just recently added to the backport.

I'll add a test that just checks that the functions runs and we don't hit any weird errors. (Of course, in our CI, I expect the list of third-party dependencies to be empty, and it'll be difficult to actually test the detection in our CI.)

Reply:
Okay, turns out there's actually no `packages_distributions` in `importlib.metadata`... so I guess the solution is to just always import this function from our fork?